corehq/apps/app_manager/util.py

```diff
@@ -692,6 +694,23 @@ def get_latest_enabled_build_for_profile(domain, profile_id):
         return get_app(domain, latest_enabled_build.build_id)
 
 
+@quickcache(['app_or_build_id'], timeout=24 * 60 * 60)
+def get_build_app_id(app_or_build_id):
+    from corehq.apps.app_manager.models import Application
+    # if its a build id, get the application id
+    app_or_release = Application.get_db().get(app_or_build_id)
+    return app_or_release.get('copy_of', app_or_build_id)
+
+
+@quickcache(['domain', 'location_id', 'app_id'], timeout=24 * 60 * 60)
+def get_latest_enabled_build_for_location(domain, location_id, app_id):
+    location = SQLLocation.active_objects.get(location_id=location_id)
+    version_to_update = location.restricted_app_releases.get(app_id)
```

Comment:
how will this work if the user's location is lower in the location hierarchy? e.g. this is a District but user location is AWC